src/identity/interaction/util/BaseInteractionCompleter.ts

```diff
@@ -0,0 +1,29 @@
+import type { InteractionResults } from 'oidc-provider';
+import type { InteractionCompleterInput } from './InteractionCompleter';
+import { InteractionCompleter } from './InteractionCompleter';
+
+/**
+ *  Creates a simple InteractionResults object based on the input parameters and injects it in the Interaction.
+ */
+export class BaseInteractionCompleter extends InteractionCompleter {
+  public async handle(input: InteractionCompleterInput): Promise<string> {
+    const now = Math.floor(Date.now() / 1000);
+    const result: InteractionResults = {
+      login: {
+        account: input.webId,
+        remember: input.shouldRemember,
+        ts: now,
+      },
+      consent: {
+        rejectedScopes: input.shouldRemember ? [] : [ 'offline_access' ],
+      },
+    };
+
+    // Identical behaviour to calling `provider.interactionResult`
```

Comment:
Not sure what this comment means or what happens here.

Reply:
I _think_ what it does is computing/redirection to a URL that captures this information; if that's correct, write that down?

Reply:
What actually happens here is that I went into the source code of the OIDC library and copied the behaviour from the `interactionResult` function there: https://github.com/panva/node-oidc-provider/blob/19b4d0daa4ca1e05acd2b5651545251fe937ff39/lib/provider.js#L243

Reason being that this way we can void having to pass the response object (since it's not needed even though the function asks for it if you already have the interaction object). So this is also here as a reminder in case the OIDC library ever chooses to change this behaviour and we upgrade to that version.

Reply:
That's clear; maybe add that?